nomad/structs/consul.go

```diff
@@ -44,6 +49,20 @@ func (c *Consul) Validate() error {
 	return nil
```

Comment:
Note we don't want to do the name validation here because this is allowed to be empty. The job admission hook will handle ensuring the cluster exists.